File: app/src/main/java/com/example/mynewapplication/javapackage/adapter/ContentListAdapter.java
package com.example.mynewapplication.javapackage.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.example.mynewapplication.R;
import com.example.mynewapplication.javapackage.ContentDetailActivity;
import com.example.mynewapplication.javapackage.model.Results;
import java.util.ArrayList;
import java.util.List;

public class ContentListAdapter extends RecyclerView.Adapter<ContentListAdapter.ContentListViewHolder> {

    private List<Results> dataList = new ArrayList<>();
    private Context context;

    public ContentListAdapter(Context context, List<Results> dataList){
        this.dataList = dataList;
        this.context = context;
    }

    @NonNull
    @Override
    public ContentListViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View view = layoutInflater.inflate(R.layout.layout_recyclerview_contentlist, parent, false);

        return new ContentListViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ContentListViewHolder holder, int position) {

        final Results model = dataList.get(position);

        holder.tv_titlecontentlist.setText(dataList.get(position).getTitle());
        holder.tv_desccontentlist.setText(dataList.get(position).getIngredients());

        Glide.with(context)
                .load(dataList.get(position).getThumbnail())
                .into(holder.iv_contentlist);

        holder.btn_contentlist.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(holder.itemView.getContext(), ContentDetailActivity.class);
                i.putExtra("href", model.getHref());
                holder.itemView.getContext().startActivity(i);
            }
        });

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(holder.itemView.getContext(), ContentDetailActivity.class);
                i.putExtra("href", model.getHref());
                holder.itemView.getContext().startActivity(i);
            }
        });
    }

    @Override
    public int getItemCount() {
        return dataList.size();
    }

    public class ContentListViewHolder extends RecyclerView.ViewHolder{

        private final ImageView iv_contentlist;
        private final TextView tv_titlecontentlist, tv_desccontentlist;
        private final Button btn_contentlist;

        public ContentListViewHolder(@NonNull View itemView) {
            super(itemView);
            iv_contentlist = itemView.findViewById(R.id.iv_contentlist);
            tv_titlecontentlist = itemView.findViewById(R.id.tv_titlecontentlist);
            tv_desccontentlist = itemView.findViewById(R.id.tv_desccontentlist);
            btn_contentlist = itemView.findViewById(R.id.btn_contentlist);

        }
    }
}

File: app/src/main/java/com/example/mynewapplication/Adapter/ContentListAdapter.kt
package com.example.mynewapplication.Adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.mynewapplication.KotlinModel.ListModel
import com.example.mynewapplication.R

class ContentListAdapter (private val data: ArrayList<ListModel>,
                          private var onItemClickListener: OnItemClickListener): RecyclerView.Adapter<ContentListAdapter.ContentListViewHolder>() {

    private var btnContent: Button? = null

    fun setOnclickListener(listener: OnItemClickListener){
        this.onItemClickListener = onItemClickListener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ContentListViewHolder {
        val inflater: LayoutInflater = LayoutInflater.from(parent.context)
        return ContentListViewHolder(inflater, parent)
    }

    override fun onBindViewHolder(holder: ContentListViewHolder, position: Int) {
        holder.bind(data[position])
        holder.itemView.setOnClickListener {
            onItemClickListener.onClick(data[position]) }
        btnContent?.setOnClickListener { onItemClickListener.onClick(data[position]) }
    }

    override fun getItemCount(): Int {
        return data.size
    }

    class ContentListViewHolder(inflater: LayoutInflater, parent: ViewGroup):
            RecyclerView.ViewHolder(inflater.inflate(R.layout.layout_recyclerview_contentlist, parent, false)) {

        private var imgView: ImageView? = null
        private var txtTitle: TextView? = null
        private var txtDesc: TextView? = null
        private var txtDetail1: TextView? = null
        private var txtDetail2: TextView? = null
        private var txtDetail3: TextView? = null
        private var btnContent: Button? = null

        init {
            imgView = itemView.findViewById(R.id.iv_contentlist)
            txtTitle = itemView.findViewById(R.id.tv_titlecontentlist)
            txtDesc = itemView.findViewById(R.id.tv_desccontentlist)
            txtDetail1 = itemView.findViewById(R.id.tv_detail_list1)
            txtDetail2 = itemView.findViewById(R.id.tv_detail_list2)
            txtDetail3 = itemView.findViewById(R.id.tv_detail_list3)
            btnContent = itemView.findViewById(R.id.btn_contentlist)
        }

        fun bind(data: ListModel){
            imgView?.setImageResource(data.imgView)
            txtTitle?.text = data.txtTitle
            txtDesc?.text = data.txtDesc
            txtDetail1?.text = data.txtDetail1
            txtDetail2?.text = data.txtDetail2
            txtDetail3?.text = data.txtDetail3
            btnContent?.text = data.btnContent
        }
    }

    interface OnItemClickListener{
        fun onClick(data: ListModel)
    }
}